README.md

```diff
@@ -1,12 +1,63 @@
 celvet
 ======
 
+**celvet** is a multitool for linting CRDs (`CustomResourceDefinition`) in
+Kubernetes. `celvet` also provides particular insight into [validation rules](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules),
+checking that they do not exceed the resource constraints provided by
+Kubernetes' API server. The tool is deliberately made to be useful both in an
+ad-hoc fashion by developers in addition to being part of a CI/CD pipeline, so
+`celvet` supports output in JSON in addition to machine-readable text.
+
+What does celvet catch?
+-----------------------
+
+* **CRD parsing errors** - celvet will early exit if it finds that a CRD
+    contains invalid YAML/JSON.
+* **Missing limits on variable-length fields** - `maxLength`, `maxItems`, etc.
+* **Validation rule parsing errors**  - if celvet has issues parsing a
+    [validation rule](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#validation-rules), it will output the errors.
+* **Exceeding resource constraints for validation rules** - if a validation
+    rule exceeds the [per-rule cost budget](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#resource-use-by-validation-functions), celvet will output an error.
+    Additionally, if the sum total of rules on the CRD exceed the per-CRD cost
+    budget, celvet will output an error there as well.
+
+Installing
+----------
+
+celvet can be installed with `go get`:
+
+    go get github.com/DangerOnTheRanger/celvet
+
 Usage
 -----
 
 ```
 celvet crd-file
 ```
 
-If `celvet` finds any linting errors, it will print them to stdout and return
-a non-zero error code.  
\ No newline at end of file
+If celvet finds any linting errors, it will print them to stdout and return
+a non-zero error code. For example, given one of the example CRDs from the [k8s documentation](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition),
+celvet will output:
+```
+string "spec.validation.openAPIV3Schema.properties[spec].properties[cronSpec]" missing maxLength
+string "spec.validation.openAPIV3Schema.properties[spec].properties[image]" missing maxLength
+```
+
+The `-o`/`--output` option can be used to change the output format to JSON if
+`json` is passed to the flag; `text` is the default option but can be
+explicitly passed. The same file but with `-o json` passed will give the
+following (the data has been pretty-printed here for readability):
+```json
+{
+  "limitErrors": [
```

Comment:
Apologies for being this up so late in the review process, but do we really want a different group for each type of error? As more and more linter rules are added, will this result a large number of different error groups that a client must know how to process?

Reply:
Also, how does a user know what "limitErrors" are and how to fix them? (same question applies to all error categories)

Reply:
I've added a list to the README of the different linter error types and opened #10 to track fixing the JSON output - fixed